### ISSUE TITLE
Type error false positive when refining impl trait in RPITIT

### ISSUE THREAD

**Issue by maxdexh** [rust-lang/rust-analyzer#22714]
**versions**: rustc 1.94 (stable, RA 2026-03-25) to rustc 1.98.0 (current nightly, rustc/RA 2026-07-02)

**editor or extension**: VSCode, nvim (rustaceanvim)

**relevant settings**: tested on default settings

Rust-Analyzer does not support refined impl trait types in RPITIT. However, since 1.94, this also causes type errors in code that passes `cargo build`:

```rust
trait Trait {
    fn method(&self) -> impl Sized;
}
#[expect(refining_impl_trait)]
impl Trait for i32 {
    fn method(&self) -> i32 {
        *self
    }
}
fn main() {
    let _: i32 = 1i32.method(); // rust-analyzer: expected i32, found impl Sized
}
```

**Comment by ChayimFriedman2** [comment]
This is blocked on us implementing RPITIT correctly.
